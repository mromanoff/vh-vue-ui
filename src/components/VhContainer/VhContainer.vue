<template>
  <div
    class="VhContainer"
    :class="{ 'VhContainer--isTop': this.isTop }"
  >
    <div class="VhContainer-main">
      <slot name="main" />
    </div>
    <aside class="VhContainer-aside u-sm-hidden u-md-hidden">
      <slot name="aside" />
    </aside>
  </div>
</template>

<script>
export default {
  props: {
    isTop: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
.VhContainer {
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  padding-left: var(--space--medium);
  padding-right: var(--space--medium);
}

.VhContainer-main {
  flex: 1 1 100%;
  width: 100%;
}

.VhContainer-aside >>> .Ad:first-child {
  margin-top: 0;
}

.VhContainer-aside:empty {
  display: none;
}

.VhContainer-main + .VhContainer-aside {
  max-width: var(--aside--width);
  flex: 0 0 var(--aside--width);
  margin-bottom: var(--space--large);
}

.VhContainer--isTop {
  margin-top: var(--space--xLarge);
}

@media (--viewport--medium) {
  .VhContainer-main {
    overflow: hidden;
    margin-bottom: 0;
    display: block;
  }
}

@media (--viewport--medium-up) {
  .VhContainer {
    flex-direction: row;
    width: var(--breakpoint--large);
    max-width: 100%;
  }
  .VhContainer-main + .VhContainer-aside {
    margin-left: calc(var(--space--medium) * 2);
    margin-bottom: 0;
  }
}
</style>
