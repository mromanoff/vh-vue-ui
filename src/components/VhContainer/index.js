import VhContainer from './VhContainer.vue';

export default VhContainer;
