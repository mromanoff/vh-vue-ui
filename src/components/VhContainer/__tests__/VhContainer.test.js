import { shallowMount } from '@vue/test-utils';
import VhContainer from '../VhContainer.vue';

describe('VhContainer.vue', () => {
  let wrapper;

  it('renders default VhContainer', () => {
    wrapper = shallowMount(VhContainer);

    expect(wrapper).toMatchSnapshot();
  });
});
