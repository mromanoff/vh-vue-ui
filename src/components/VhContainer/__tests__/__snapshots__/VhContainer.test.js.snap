// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhContainer.vue renders default VhContainer 1`] = `
<div class="Container">
  <div class="Container-main"></div>
  <aside class="Container-aside u-sm-hidden u-md-hidden"></aside>
</div>
`;
